bodhi/server/validators.py

```diff
@@ -989,6 +989,39 @@ def validate_comment_id(request, **kwargs):
         request.errors.status = HTTPNotFound.code
 
 
+def validate_new_comment(request, **kwargs):
+    """
+    Unflatten the content of new comment to accord with 'SaveCommentSchema' data structure.
+
+    For list of bug_feedback and testcase_feedback, the request object has data in format as:
+        "bug_feedback.0.bug_id": "1234", "bug_feedback.0.karma": "1"
+    which, in order for schema to be validated, needs to be converted as follows:
+        bug_feedback: [ { "bug_id": "1234", "karma": "1" }]
+
+    Args:
+        request (pyramid.util.Request): The current request.
+        kwargs (dict): The kwargs of the related service definition. Unused.
+    """
+    # In-method import used because there is mutual referencing between
+    # 'bodhi.server.schemas' and 'bodhi.server.validators'
+    from bodhi.server.schemas import SaveCommentSchema
+
+    content = str(request.body)
```

Comment:
I think here you could use the [request.json ](https://docs.pylonsproject.org/projects/pyramid/en/latest/api/request.html#pyramid.request.Request.json)attribute that will give you directly a dictionary to work with.

Reply:
But that won't be set in test environment. So should we set that also in testing or make a decision based logic of whether parameters are set?

Reply:
in the test instead of setting `request.body` you should be able to set `request.json` instead.

Reply:
```request.json``` stores every key-value pair in ```unicode``` format. However, the ```colander``` parser required it to be ```str```. So I can either convert each key-value of ```request.json``` into ```str``` or I can use the method above. Which one would be more efficient according to you?